= class KeyError < IndexError

[[m:Hash#fetch]] などで key に対応する value がない場合に発生します。

Ruby 1.8 以前では同様の場面で [[c:IndexError]] が発生していました。
互換性のため、[[c:KeyError]] は [[c:IndexError]] のサブクラスになっています。

#@since 2.5.0
== Instance Methods

--- receiver -> object

KeyError の原因となったメソッド呼び出しのレシーバを返します。

@raise ArgumentError レシーバが設定されていない時に発生します。

例:

  h = Hash.new
  begin
    h.fetch('gumby'*20)
  rescue KeyError => e
    p e.message             # => "key not found: \"gumbygumbygumbygumbygumbygumbygumbygumbygumbygumbygumbygumbyg..."
    p h.equal?(e.receiver)  # => true
  end

--- key -> object

KeyError の原因となったメソッド呼び出しのキーを返します。

@raise ArgumentError キーが設定されていない時に発生します。

例:

  h = Hash.new
  begin
    h.fetch('gumby'*20)
  rescue KeyError => e
    p e.message            # => "key not found: \"gumbygumbygumbygumbygumbygumbygumbygumbygumbygumbygumbygumbyg..."
    p 'gumby'*20 == e.key  # => true
  end
#@end
